app/models/setting/content.rb

```diff
@@ -92,6 +92,8 @@ def self.default_settings
                N_('Default Custom Repository download policy'), nil, :collection => download_policies),
       self.set('default_redhat_download_policy', N_("Default download policy for enabled Red Hat repositories (either 'immediate' or 'on_demand')"), "on_demand",
                        N_('Default Red Hat Repository download policy'), nil, :collection => download_policies),
+      self.set('default_deb_download_policy', N_("Default download policy for enabled repositories of type 'deb' (either 'immediate' or 'on_demand')"), "immediate",
+                       N_('Default Debian content download policy'), nil, :collection => download_policies),
```

Comment:
Is there a reason to not just use 'default_download_policy' for deb as well?  There's nothing about that is specific to yum

Reply:
We consider `ondemand` policy as problematic. Maybe @m-bucher can elaborate on this.

Reply:
I currently suspect that it depends on the repository, whether using `on_demand` is a good idea.
If the repository does remove packages from the pool that are no longer in the current package-indizes, older versions that were synced in katello with `on_demand` will no longer work.
It is not a problem, if the upstream repository's package-pool only gets new packages and never looses older ones.

The first case might be especially problematic if the repo-versions are Content-View versions unless the packages are synced, when a Content-View version is created from the repository.
I might lack some deeper knowledge about how `on_demand` really works :sweat_smile:

Reply:
@m-bucher do you think it needs to be a different setting entirely?  currently 'default_download_policy' defaults to immediate for similar reasons you mentioned (if older packages are removed from upstream repos, it can break content views).  So we default custom repos to immediate as well.

Reply:
In that case I think we can get away with keeping it in one setting and revisit it if users demand it.

It may depend on how high the percentages are of repositories removing old packages versus repos that do not.
My gut tells me deb-repos in the wild tend to be more `remove old packages`.
Not a clue about what kind is used more often with yum-repos.
But then again I do not have reliable data for deb-repos either, just a feeling :innocent:

Reply:
okay, how about we start off using the same setting, and we can always split it out later on.  I'd rather err on the side of fewer settings.  @nadjaheitmann do you mind updating the pr to just use  'default_download_policy' ?

Reply:
So it will be just the migration, then?

Reply:
migration plus the chnage to app/models/katello/glue/pulp/repos.rb to also set it for dep repos

Reply:
Really change in repos.rb? If I look in https://github.com/Katello/katello/blob/d2c20af6911adb02ca9005350a9f2c1ad1f8140b/app/models/katello/root_repository.rb#L23, it looks like it is already checking for `deb` repos in https://github.com/Katello/katello/blob/d4f2976f4d28e574d291dcbb793afa02b7b8a363/app/models/katello/glue/pulp/repos.rb#L135.

Reply:
oh right, good call!  just the migration then!

Reply:
At least I don't have to close the whole PR because it became obsolete ;)

Reply:
Updated the PR and also the redmine issue.